core/go/internal/publictxmgr/transaction_manager.go

```diff
@@ -757,7 +757,7 @@ func (ptm *pubTxManager) MatchUpdateConfirmedTransactions(ctx context.Context, d
 	var lookups []*bindingsMatchingSubmission
 	err := dbTX.DB().
 		Table("public_txn_bindings").
-		Select(`"transaction"`, `"tx_type"`, `"Submission"."pub_txn_id"`, `"Submission"."tx_hash"`).
+		Select(`"transaction"`, "sender", "contract_address", `"tx_type"`, `"Submission"."pub_txn_id"`, `"Submission"."tx_hash"`).
```

Comment:
This was critical - meant `Sender` wasn't set (for cross-node propagation), and `ContractAddress` for forcing a reset of the in-memory state of the privacy domain when transactions fail after dispatch.